probes/hasApprovedLicense/def.yml

```diff
@@ -0,0 +1,27 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: hasApprovedLicense
+short: Check that the project has a license file of approved format. An approved format is a file named either LICENSE, LICENCE, COPYING or COPYRIGHT with the file extension .html, .txt, or .md.
```

Comment:
I don't think that's what "approved" means. I think it means it's a license that lets end-users use the code without copyright, e.g. Apache. Double check the code to confirm

Reply:
I believe "approved" in this case is whether or not it's a OSI recognized license:
https://github.com/ossf/scorecard/blob/5b8a33ddea7c923ec05843bd18fa1dbe5612c6fc/checks/evaluation/license.go#L59-L60

https://opensource.org/licenses/

Reply:
so we should update the text of thr probe. Right now it says `approved format is a file named ...` but that's not what an approve license is, iiuc

Reply:
Updated the text in https://github.com/ossf/scorecard/pull/3465/commits/4d4a924c0133fbb2477e48e81159a82617a9c5fb.